infra/main.tf

```diff
@@ -23,14 +24,145 @@ resource "aws_security_group" "allow_ssh" {
     protocol    = "tcp"
     cidr_blocks = ["0.0.0.0/0"]
   }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_iam_policy" "poc_ec2_policy" {
+  name        = var.policy_name
+  description = "Policy to provide permissions to poc-ec2 instance"
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "secretsmanager:GetResourcePolicy",
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:ListSecretVersionIds",
+          "secretsmanager:ListSecrets"
+        ],
+        "Resource" : [
+          "*"
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role" "poc_ec2_role" {
+  name = var.iam_role_name
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Sid" : "",
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "ec2.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy_attachment" "poc_attach_policy_to_role" {
+  name       = "ec2Attachment"
+  roles      = [aws_iam_role.poc_ec2_role.name]
+  policy_arn = aws_iam_policy.poc_ec2_policy.arn
+}
+
+resource "aws_iam_instance_profile" "poc_ec2_profile" {
+  name = var.profile_name
+  role = aws_iam_role.poc_ec2_role.name
+}
+
+
+# resource "tls_private_key" "poc_rsa" {
+# algorithm = "RSA"
+# rsa_bits  = 4096
+# }
+
+# resource "aws_key_pair" "poc_key_pair" {
+# key_name = var.ssh_key_name
+# public_key = tls_private_key.poc_rsa.public_key_openssh
+# }
+# resource "local_file" "tf-key" {
+# content  = tls_private_key.rsa.private_key_pem
+# filename = "tf-key-pair"
+# }
+
 
-  tags = var.tags
+
+
+
+# Generate a private TLS key using RSA
```

Comment:
Same. Comments are used only to explain why the following code was used. Not what it does.